### PR TITLE
add handlers for channels

### PR DESCRIPTION
**Pull request by neha-Gupta1**
add Handlers interface for Channels. 

#### Does this PR need a docs update or release note?
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature

#### Issue(s)

* #<issue>

### CODE REVIEW

File: src/internal/m365/collection/groups/handler.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type BackupMessagesHandler interface {
+	GetMessage(ctx context.Context, teamID, channelID, itemID string) (models.ChatMessageable, error)
+	NewMessagePager(teamID, channelID string) api.MessageItemDeltaEnumerator
+	GetChannel(ctx context.Context, teamID, channelID string) (models.Channelable, error)
+	NewChannelPager(teamID, channelID string) api.ChannelItemDeltaEnumerator
+	GetReply(ctx context.Context, teamID, channelID, messageID string) (serialization.Parsable, error)
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/groups/handler.go:17]: Do we need a `GetReply` pager?
```suggestion
	GetReplyByID(ctx context.Context, teamID, channelID, messageID string) (serialization.Parsable, error)
```

File: src/internal/m365/collection/groups/handler.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type BackupMessagesHandler interface {
+	GetMessage(ctx context.Context, teamID, channelID, itemID string) (models.ChatMessageable, error)
+	NewMessagePager(teamID, channelID string) api.MessageItemDeltaEnumerator
+	GetChannel(ctx context.Context, teamID, channelID string) (models.Channelable, error)
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/groups/handler.go:15]:
```suggestion
	GetChannelByID(ctx context.Context, teamID, channelID string) (models.Channelable, error)
```

File: src/internal/m365/collection/groups/handler.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	"context"
+
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type BackupMessagesHandler interface {
+	GetMessage(ctx context.Context, teamID, channelID, itemID string) (models.ChatMessageable, error)
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/groups/handler.go:13]:
```suggestion
	GetMessageByID(ctx context.Context, teamID, channelID, itemID string) (models.ChatMessageable, error)
```

File: src/internal/m365/collection/groups/handler.go (hunk 1 above)
Comment (neha-Gupta1) [reply src/internal/m365/collection/groups/handler.go:17]: I will add `GetReply` pager if required while implementing enumerator.

**Pull request merged**